src/measure_theory/outer_measure.lean

```diff
@@ -96,6 +97,48 @@ protected lemma union (m : outer_measure α) (s₁ s₂ : set α) :
   m (s₁ ∪ s₂) ≤ m s₁ + m s₂ :=
 rel_sup_add m m.empty (≤) m.Union_nat s₁ s₂
 
+/-- If `s : ι → set α` is a sequence of sets, `S = ⋃ n, s n`, and `m (S \ s n)` tends to zero along
+some nontrivial filter (usually `at_top` on `α = ℕ`), then `m S = ⨆ n, s n`. -/
+lemma Union_of_tendsto_zero {ι} (m : outer_measure α) {s : ι → set α}
+  (l : filter ι) [ne_bot l] (h0 : tendsto (λ k, m ((⋃ n, s n) \ s k)) l (𝓝 0)) :
+  m (⋃ n, s n) = ⨆ n, m (s n) :=
+begin
+  set S := ⋃ n, s n,
+  set M := ⨆ n, m (s n),
+  have hsS : ∀ {k}, s k ⊆ S, from λ k, subset_Union _ _,
+  refine le_antisymm _ (supr_le $ λ n, m.mono hsS),
+  have A : ∀ k, m S ≤ M + m (S \ s k), from λ k,
+  calc m S = m (s k ∪ S \ s k) : by rw [union_diff_self, union_eq_self_of_subset_left hsS]
+  ... ≤ m (s k) + m (S \ s k) : m.union _ _
+  ... ≤ M + m (S \ s k) : add_le_add_right (le_supr _ k) _,
+  have B : tendsto (λ k, M + m (S \ s k)) l (𝓝 (M + 0)), from tendsto_const_nhds.add h0,
+  rw add_zero at B,
+  exact ge_of_tendsto' B A
+end
+
+/-- If `s : ℕ → set α` is a monotone sequence of sets such that `∑' k, m (s (k + 1) \ s k) ≠ ∞`,
+then `m (⋃ n, s n) = ⨆ n, m (s n)`. -/
+lemma Union_nat_of_monotone_of_tsum_ne_top (m : outer_measure α) {s : ℕ → set α}
+  (h_mono : ∀ n, s n ⊆ s (n + 1)) (h0 : ∑' k, m (s (k + 1) \ s k) ≠ ∞) :
+  m (⋃ n, s n) = ⨆ n, m (s n) :=
+begin
+  refine m.Union_of_tendsto_zero at_top _,
+  refine tendsto_nhds_bot_mono' (ennreal.tendsto_sum_nat_add _ h0) (λ n, _),
+  refine (m.mono _).trans (m.Union _),
+  /- Current goal: `(⋃ k, s k) \ s n ⊆ ⋃ k, s (k + n + 1) \ s (k + n)`
+  Should we add this to `data/set/*`? -/
```

Comment:
This doesn't seem of general enough interest to be moved to `data/set`. I'd just erase the second line of the comment.